### PR TITLE
[db-sync] Optionally read DB credentials from env vars

### PR DESCRIPTION
**Pull request by geropl**
## Description
Optionally read DB credentials from env vars ([commit without formatting](https://github.com/gitpod-io/gitpod/pull/13573/commits/15bb1f3857868132557550d326369bd1e889292a)).

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

File: components/ee/db-sync/src/commands.ts [hunk 1]
```diff
@@ -39,40 +39,63 @@ export class RunCommand implements ICommand {
...
         const config = JSON.parse(readFileSync(args.config).toString()) as ReplicationConfig;
 
-        if(config.roundRobin) {
-            if(config.source) {
+        if (process.env.DB_PASSWORD) {
+            console.info("Found DB_PASSWORD env var, using it to establish connections.");
+            if (config.source) {
+                config.source.password = process.env.DB_PASSWORD;
+            }
+            for (const target of config.targets) {
```
Comment (easyCZ) [inline components/ee/db-sync/src/commands.ts:54]: Looks like this handling of multiple targets was not present in here before. This PR doesn't seem to introduce the targets either. Is this just an oversight?

Comment (easyCZ) [reply components/ee/db-sync/src/commands.ts:54]: And if it should be happening even when the var is not set through ENV, we could hoist it outside of the `if (process.env.DB_*)`

Comment (geropl) [reply components/ee/db-sync/src/commands.ts:54]: I'm not sure I follow. [Besides the formatting](https://github.com/gitpod-io/gitpod/pull/13573/commits/15bb1f3857868132557550d326369bd1e889292a), we just check both env vars now, and use them to override all other DB usernames/passwords.

We effectively only use `roundRobin`, so `source` is not strictly required, but found it cleaner to add it as well.
Does that make sense? :thinking:

Comment (easyCZ) [reply components/ee/db-sync/src/commands.ts:54]: I was confused that we override all the targets which was not done originally, but I think I've wrapped my head around it  - it was statically provided before and now it's dynamic due to the ENV var override

**Review by easyCZ — APPROVED** [review]

**Pull request merged**